lib/beaker/hypervisor/aws_sdk.rb

```diff
@@ -649,6 +649,14 @@ def set_hostnames
           else
             next if host['platform'] =~ /netscaler/
             host.exec(Command.new("hostname #{host.name}"))
+            if host['vmname'] =~ /^amazon/
```

Comment:
Yes, the process for setting persistent hostname seems to be peculiar to the Amazon Linux builds:

http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/set-hostname.html

(if I'm following along...)